scraping/scraping/spiders/amazon.py

```diff
@@ -0,0 +1,70 @@
+from logging import getLogger
+from typing import Iterator
+
+from scrapy_splash import SplashJsonResponse, SplashRequest
+
+from ..splash import scroll_end_of_page_script
+from ._base import BaseSpider
+
+logger = getLogger(__name__)
+
+
+class AmazonSpider(BaseSpider):
+    name = "amazon"
+    allowed_domains = ["amazon.de"]
+
+    def parse_SERP(self, response: SplashJsonResponse) -> Iterator[SplashRequest]:
+        """
+        The `Scrapy` framework executes this method.
+
+        Yields:
+            Iterator[SplashRequest]: Requests that will be performed to scrap each product page
+        """
+        # Save HTML to database
+        self._save_SERP(response)
+        urls = response.css("div.a-row.a-size-base.a-color-base a::attr(href)").getall()
+        prices = response.css(
+            "div.a-row.a-size-base.a-color-base span.a-price-whole::text"
+        ).getall()
+        products = dict(zip(urls, prices))
+
+        # Yield request for each product
+        logger.info(f"Number of products to be scraped {len(products)}")
+        for key, value in products.items():
+            if "refinements=p_n_cpf_eligible" in key:
+                yield SplashRequest(
+                    url=f"https://www.amazon.de{key}",
+                    callback=self.parse_PRODUCT,
+                    meta={
+                        "amazon_price": value
+                    },  # customized meta_information working used just in amazon
+                    endpoint="execute",
+                    priority=1,  # higher priority than SERP
+                    args={  # passed to Splash HTTP API
+                        "wait": self.request_timeout,
+                        "lua_source": scroll_end_of_page_script,
```

Comment:
Just out of curiosity, did you try the `minimal_script`?

Reply:
I tried a few times successfully but not enough to say if it's working fine for all pages, should I implement it?

Reply:
Maybe let's finalize everything and then try with each script and compare the results.